### PR TITLE
Simplify the random bits generator

### PR DESCRIPTION
**Pull request by martinthomson**
This won't get to the real problem we're dealing with here, but it should avoid most of the overhead in tests.

### CODE REVIEW

**Comment by martinthomson** [comment]
Before:
```
rows 100 benchmark complete after 5.0614614s
```

After:
```
rows 100 benchmark complete after 3.7972748s
```

**Comment by martinthomson** [comment]
For `cargo test`

Before:
```
real    0m26.459s
user    1m40.707s
sys     0m8.048s
```

After:
```
real    0m9.582s
user    0m51.094s
sys     0m4.170s
```

ლ(╹◡╹ლ)

File: src/protocol/boolean/random_bits_generator.rs [hunk 1]
```diff
@@ -4,142 +4,21 @@ use crate::ff::Field;
...
+    marker::PhantomData,
+    sync::atomic::{AtomicU32, AtomicUsize, Ordering},
+};
 
 /// A struct that pre-generates and buffers random sharings of bits from the
 /// `SolvedBits` protocol. Any protocol who wish to use a random-bits can draw
-/// one by calling `take_one()`. It will call `SolvedBits` once the stock falls
-/// below `REFILL_THRESHOLD` until it fills up the empty slots.
+/// one by calling `take_one()`. Making a generator spawns a task that generates
+/// new values asynchronously up to the given capacity.
```
Comment (taikiy) [inline src/protocol/boolean/random_bits_generator.rs:15]: The code seems to just generate one `RandomBitsShare` for each `generate()` call?

**Review by taikiy — APPROVED** [review]
Thanks for this!

File: src/protocol/boolean/random_bits_generator.rs (hunk 1 above)
Comment (martinthomson) [reply src/protocol/boolean/random_bits_generator.rs]: Yeah, I fixed that already.  Failed to push.

**Pull request merged**